test/kt/two_pass_scan.cpp

```diff
@@ -67,30 +67,58 @@ generate_scan_data(T* input, std::size_t size, std::uint32_t seed)
     }
 }
 
-template <typename T, sycl::usm::alloc _alloc_type, typename BinOp, typename KernelParam>
+template <typename T, sycl::usm::alloc _alloc_type, typename BinOp, typename KernelParam1, typename KernelParam2>
 void
-test_usm(sycl::queue q, std::size_t size, BinOp bin_op, KernelParam param)
+test_usm(sycl::queue q, std::size_t size, BinOp bin_op, KernelParam1 param1, KernelParam2 param2)
```

Comment:
We probably want to add exclusive scan for the other input types eventually, although its not a blocker for this PR.

Reply:
I agree it may be good. We will probably want to cut down on some testing somewhere though or maybe added a short / long mode for testing. With integrated graphics, running this test is already taking 400+ seconds on my side.